### ISSUE TITLE
MetricFilter's matches returns true if Includes are specified and Excludes are not

### ISSUE THREAD

**Issue by qinfchen** [dropwizard/dropwizard#711]
If Includes are not empty and excludes are empty, this should be the includes-only case that 
no metrics should be reported, except those with a name listed in includes.

But !getExcludes().contains(name) would always return true which ends up reporting all the metrics.

**Comment by eepstein** [comment]
I think the bug fix also has a minor bug in it.  In particular, "case 4" of the unit tests seems to be different that one would imagine with a pair of exclude and include filters.

Normally with such a pair the logic is:

     include =  !isExcluded && isIncluded

So the exclusions are applied first and then anything that makes it past the exclusions and is in the explicit set of inclusions gets included.  But the logic in place now has case 4's includes overriding the excludes, which given that the default behavior is including-everything, is strange.  One has explicitly excluding something and then support explicitly overriding the exclusion.  That's double the effort when all that's needed is: don't put the thing in the exclusion list in the first place.  

In short, exclusions should be applied first, and the items that aren't excluded are now the set from which items may be explicitly included.  No explicit inclusion means: include everything not explicitly excluded (case 3).  Case 4, as coded, is inverted.
